### PR TITLE
Upgrade to ocamlformat 0.26.0

### PR DESCRIPTION
**Pull request by Julow**
The aim of this commit is to gather feedback.

Changelog can be found here: https://github.com/ocaml-ppx/ocamlformat/blob/main/CHANGES.md

### CODE REVIEW

File: doc/examples/markup.mli [hunk 1]
```diff
@@ -52,46 +52,43 @@ module type Foo = sig
...
   (** This comment is associated to bar. *)
 
-  class cl :
-    object
-
-      (** Interesting information about cl *)
-    end
+  class cl : object
+    
+(** Interesting information about cl *)
```
Comment (jonludlam) [inline doc/examples/markup.mli:57]: This seems an unfortunate change.

File: src/document/url.ml [hunk 2]
```diff
@@ -253,7 +253,7 @@ module Anchor = struct
     | None -> assert false (* We got a root, should never happen *)
     | Some page ->
         let anchor = Printf.sprintf "%s-%s" (Path.string_of_kind kind) name in
-        { page; anchor; kind = (kind :> kind) }
+        { page; anchor; kind :> kind }
```
Comment (jonludlam) [inline src/document/url.ml:256]: Didn't know this was allowed!

**Review by jonludlam — COMMENTED** [review]
Looks pretty good to me, with that one exception.

File: src/document/url.ml (hunk 2 above)
Comment (Julow) [reply src/document/url.ml:256]: Not sure about the compatibility. I'll check.

File: doc/examples/markup.mli (hunk 1 above)
Comment (Julow) [reply doc/examples/markup.mli:57]: Good catch!

File: src/document/url.ml (hunk 2 above)
Comment (Julow) [reply src/document/url.ml:256]: This is a syntax error on 4.02. This is a bug!

Comment (Julow) [reply src/document/url.ml:256]: Fixed in https://github.com/ocaml-ppx/ocamlformat/pull/2388

File: doc/examples/markup.mli (hunk 1 above)
Comment (Julow) [reply doc/examples/markup.mli:57]: Fixed in https://github.com/ocaml-ppx/ocamlformat/pull/2387

**Comment by Julow** [comment]
I updated the preview with the bugs fixed.
Is the `.git-blame-ignore-revs` file a welcome addition ?

**Comment by trefis** [comment]
> Is the `.git-blame-ignore-revs` file a welcome addition ?
Yes!

**Comment by Julow** [comment]
OCamlformat 0.26.0 has been released! This PR can be merged (not squashed due to `.git-blame-ignore-revs`) or closed if it's a bit too early to upgrade.

**Review by gpetiot — APPROVED** [review]
Everything looks good to me.

**Comment by Julow** [comment]
OCamlformat will be released with compatibility with 5.1 in a few weeks. Let's skip 0.26.0.

**Pull request closed without merging**